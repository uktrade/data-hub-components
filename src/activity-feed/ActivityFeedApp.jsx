import React from 'react'
import axios from 'axios'
import ActivityFeed from './ActivityFeed'
import PropTypes from 'prop-types'

export default class ActivityFeedApp extends React.Component {
  static propTypes = {
    apiEndpoint: PropTypes.string.isRequired,
    queryParams: PropTypes.object,
    addContentText: PropTypes.string,
    addContentLink: PropTypes.string,
    render: PropTypes.func,
  }

  static defaultProps = {
    queryParams: {},
  }

  constructor (props) {
    super(props)
    this.state = {
      activities: [],
      isLoading: true,
      hasMore: true,
      offset: 0,
      total: 0,
      error: false,
    }

    this.onLoadMore = this.onLoadMore.bind(this)
  }

  async componentDidMount () {
    await this.onLoadMore()
  }

  static async fetchActivities (apiEndpoint, offset, limit, queryParams = {}) {
    const params = {
      size: limit,
      from: offset,
      ...queryParams,
    }

    const { data } = await axios.get(apiEndpoint, { params })
    const { total, hits } = data.hits

    return {
      total: total,
      activities: hits.map(hit => hit._source),
    }
  }

  async onLoadMore () {
    const { offset } = this.state
    const { apiEndpoint, queryParams } = this.props
    const limit = 20

    this.setState({
      isLoading: true,
    })

    try {
      const { activities, total } = await ActivityFeedApp.fetchActivities(apiEndpoint, offset, limit, queryParams)
      const allActivities = this.state.activities.concat(activities)

      this.setState({
        activities: allActivities,
        isLoading: false,
        hasMore: total > allActivities.length,
        offset: offset + limit,
        total,
      })
    } catch (e) {
      this.setState({
        isLoading: false,
        hasMore: false,
        error: true,
      })
    }
  }

  render () {
    const { activities, isLoading, hasMore, total, error } = this.state
    const { addContentText, addContentLink, render } = this.props

    const isEmptyFeed = activities.length === 0 && !hasMore

    return (
      <ActivityFeed
        totalActivities={total}
        addContentText={addContentText}
        addContentLink={addContentLink}
        activities={activities}
        hasMore={hasMore}
        onLoadMore={this.onLoadMore}
        isLoading={isLoading}>
        {isEmptyFeed && !error && <div>There are no activities to show.</div>}
        {error && <div>Error occurred while loading activities.</div>}
        {render && render(this.state, this.props)}
      </ActivityFeed>
    )
  }
}
