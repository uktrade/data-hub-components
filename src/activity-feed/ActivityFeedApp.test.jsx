import React from 'react'
import renderer from 'react-test-renderer'
import nock from 'nock'

import ActivityFeedApp from './ActivityFeedApp'
import esResults from '../../fixtures/activity_feed/activity-feed-from-es'

describe('ActivityFeedApp', () => {
  test('fetches activity feed', async (done) => {
    const companyId = '0f5216e0-849f-11e6-ae22-56b6b6499611'
    const addContentLink = `/companies/${companyId}/interactions/create`
    const apiBaseUrl = 'http://nock:3000'
    const apiPath = `/companies/${companyId}/activity-feed/data`
    const apiEndpoint = apiBaseUrl + apiPath

    nock(apiBaseUrl)
      .get(/companies.*$/)
      .reply(200, esResults)

    renderer.create(
      <ActivityFeedApp
        addContentText="Add interaction"
        addContentLink={addContentLink}
        apiEndpoint={apiEndpoint}
        render={(state) => {
          if (state.error) {
            throw Error('Exception raised.')
          }
          if (!state.isLoading) {
            expect(state).toMatchSnapshot()
            done()
          }
        }}
      />,
    )
  })
})
