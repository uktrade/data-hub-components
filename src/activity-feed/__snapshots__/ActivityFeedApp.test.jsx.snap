// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActivityFeedApp fetches activity feed 1`] = `
Object {
  "activities": Array [
    Object {
      "generator": Object {
        "name": "dit:dataHub",
        "type": "Application",
      },
      "id": 99,
      "object": Object {
        "attributedTo": Array [
          Object {
            "dit:companiesHouseNumber": "987654321",
            "dit:dunsNumber": "123456789",
            "id": "dit:DataHubCompany:a4ead59f-dfc8-e511-a5e6-e4115bead28a",
            "name": "Marco Test",
            "type": Array [
              "Organization",
              "dit:Company",
            ],
          },
          Object {
            "dit:emailAddress": "john@example.com",
            "id": "dit:DataHubContact:04cadecd-2de4-4cda-9fda-59db93495dba",
            "name": "John Doe",
            "type": Array [
              "Person",
              "dit:Contact",
            ],
          },
          Object {
            "dit:emailAddress": "mariobros@dit.com",
            "dit:team": Object {
              "id": "dit:DataHubTeam:3a48318c-9698-e211-a939-e4115bead28a",
              "name": "Digital Data Hub - Live Service",
              "type": Array [
                "Group",
                "dit:Team",
              ],
            },
            "id": "dit:DataHubAdviser:2dc3c098-2f51-43c5-8767-752e8e5065b5",
            "name": "Mario Bros",
            "type": Array [
              "Person",
              "dit:Adviser",
            ],
          },
        ],
        "dit:archived": false,
        "dit:communicationChannel": Object {
          "name": "Email/Website",
        },
        "dit:service": Object {
          "name": "Account Management",
        },
        "dit:status": "complete",
        "dit:subject": "Invited to Technology Roundtable 99",
        "id": "dit:DataHubInteraction:a066cc49-b679-4cbe-93c5-3ceba2f067ec",
        "startTime": "2019-05-02T08:43:40.904649+00:00",
        "type": Array [
          "dit:Event",
          "dit:Interaction",
        ],
        "url": "https://example.com/companies/a4ead59f-dfc8-e511-a5e6-e4115bead28a/interactions/a066cc49-b679-4cbe-93c5-3ceba2f067ec",
      },
      "published": "2019-08-10T09:43:40.904+01:00",
      "type": "Announce",
    },
    Object {
      "generator": Object {
        "name": "dit:dataHub",
        "type": "Application",
      },
      "id": 98,
      "object": Object {
        "attributedTo": Array [
          Object {
            "dit:companiesHouseNumber": "987654321",
            "dit:dunsNumber": "123456789",
            "id": "dit:DataHubCompany:a4ead59f-dfc8-e511-a5e6-e4115bead28a",
            "name": "Marco Test",
            "type": Array [
              "Organization",
              "dit:Company",
            ],
          },
          Object {
            "dit:emailAddress": "john@example.com",
            "id": "dit:DataHubContact:04cadecd-2de4-4cda-9fda-59db93495dba",
            "name": "John Doe",
            "type": Array [
              "Person",
              "dit:Contact",
            ],
          },
          Object {
            "dit:emailAddress": "mariobros@dit.com",
            "dit:team": Object {
              "id": "dit:DataHubTeam:3a48318c-9698-e211-a939-e4115bead28a",
              "name": "Digital Data Hub - Live Service",
              "type": Array [
                "Group",
                "dit:Team",
              ],
            },
            "id": "dit:DataHubAdviser:2dc3c098-2f51-43c5-8767-752e8e5065b5",
            "name": "Mario Bros",
            "type": Array [
              "Person",
              "dit:Adviser",
            ],
          },
        ],
        "dit:archived": false,
        "dit:communicationChannel": Object {
          "name": "Email/Website",
        },
        "dit:service": Object {
          "name": "Account Management",
        },
        "dit:status": "complete",
        "dit:subject": "Invited to Technology Roundtable 98",
        "id": "dit:DataHubInteraction:a066cc49-b679-4cbe-93c5-3ceba2f067ec",
        "startTime": "2019-05-02T08:43:40.904649+00:00",
        "type": Array [
          "dit:Event",
          "dit:Interaction",
        ],
        "url": "https://example.com/companies/a4ead59f-dfc8-e511-a5e6-e4115bead28a/interactions/a066cc49-b679-4cbe-93c5-3ceba2f067ec",
      },
      "published": "2019-08-09T09:43:40.904+01:00",
      "type": "Announce",
    },
    Object {
      "generator": Object {
        "name": "dit:dataHub",
        "type": "Application",
      },
      "id": 97,
      "object": Object {
        "attributedTo": Array [
          Object {
            "dit:companiesHouseNumber": "987654321",
            "dit:dunsNumber": "123456789",
            "id": "dit:DataHubCompany:a4ead59f-dfc8-e511-a5e6-e4115bead28a",
            "name": "Marco Test",
            "type": Array [
              "Organization",
              "dit:Company",
            ],
          },
          Object {
            "dit:emailAddress": "john@example.com",
            "id": "dit:DataHubContact:04cadecd-2de4-4cda-9fda-59db93495dba",
            "name": "John Doe",
            "type": Array [
              "Person",
              "dit:Contact",
            ],
          },
          Object {
            "dit:emailAddress": "mariobros@dit.com",
            "dit:team": Object {
              "id": "dit:DataHubTeam:3a48318c-9698-e211-a939-e4115bead28a",
              "name": "Digital Data Hub - Live Service",
              "type": Array [
                "Group",
                "dit:Team",
              ],
            },
            "id": "dit:DataHubAdviser:2dc3c098-2f51-43c5-8767-752e8e5065b5",
            "name": "Mario Bros",
            "type": Array [
              "Person",
              "dit:Adviser",
            ],
          },
        ],
        "dit:archived": false,
        "dit:communicationChannel": Object {
          "name": "Email/Website",
        },
        "dit:service": Object {
          "name": "Account Management",
        },
        "dit:status": "complete",
        "dit:subject": "Invited to Technology Roundtable 97",
        "id": "dit:DataHubInteraction:a066cc49-b679-4cbe-93c5-3ceba2f067ec",
        "startTime": "2019-05-02T08:43:40.904649+00:00",
        "type": Array [
          "dit:Event",
          "dit:Interaction",
        ],
        "url": "https://example.com/companies/a4ead59f-dfc8-e511-a5e6-e4115bead28a/interactions/a066cc49-b679-4cbe-93c5-3ceba2f067ec",
      },
      "published": "2019-08-08T09:43:40.904+01:00",
      "type": "Announce",
    },
    Object {
      "generator": Object {
        "name": "dit:dataHub",
        "type": "Application",
      },
      "id": 96,
      "object": Object {
        "attributedTo": Array [
          Object {
            "dit:companiesHouseNumber": "987654321",
            "dit:dunsNumber": "123456789",
            "id": "dit:DataHubCompany:a4ead59f-dfc8-e511-a5e6-e4115bead28a",
            "name": "Marco Test",
            "type": Array [
              "Organization",
              "dit:Company",
            ],
          },
          Object {
            "dit:emailAddress": "john@example.com",
            "id": "dit:DataHubContact:04cadecd-2de4-4cda-9fda-59db93495dba",
            "name": "John Doe",
            "type": Array [
              "Person",
              "dit:Contact",
            ],
          },
          Object {
            "dit:emailAddress": "mariobros@dit.com",
            "dit:team": Object {
              "id": "dit:DataHubTeam:3a48318c-9698-e211-a939-e4115bead28a",
              "name": "Digital Data Hub - Live Service",
              "type": Array [
                "Group",
                "dit:Team",
              ],
            },
            "id": "dit:DataHubAdviser:2dc3c098-2f51-43c5-8767-752e8e5065b5",
            "name": "Mario Bros",
            "type": Array [
              "Person",
              "dit:Adviser",
            ],
          },
        ],
        "dit:archived": false,
        "dit:communicationChannel": Object {
          "name": "Email/Website",
        },
        "dit:service": Object {
          "name": "Account Management",
        },
        "dit:status": "complete",
        "dit:subject": "Invited to Technology Roundtable 96",
        "id": "dit:DataHubInteraction:a066cc49-b679-4cbe-93c5-3ceba2f067ec",
        "startTime": "2019-05-02T08:43:40.904649+00:00",
        "type": Array [
          "dit:Event",
          "dit:Interaction",
        ],
        "url": "https://example.com/companies/a4ead59f-dfc8-e511-a5e6-e4115bead28a/interactions/a066cc49-b679-4cbe-93c5-3ceba2f067ec",
      },
      "published": "2019-08-07T09:43:40.904+01:00",
      "type": "Announce",
    },
    Object {
      "generator": Object {
        "name": "dit:dataHub",
        "type": "Application",
      },
      "id": 95,
      "object": Object {
        "attributedTo": Array [
          Object {
            "dit:companiesHouseNumber": "987654321",
            "dit:dunsNumber": "123456789",
            "id": "dit:DataHubCompany:a4ead59f-dfc8-e511-a5e6-e4115bead28a",
            "name": "Marco Test",
            "type": Array [
              "Organization",
              "dit:Company",
            ],
          },
          Object {
            "dit:emailAddress": "john@example.com",
            "id": "dit:DataHubContact:04cadecd-2de4-4cda-9fda-59db93495dba",
            "name": "John Doe",
            "type": Array [
              "Person",
              "dit:Contact",
            ],
          },
          Object {
            "dit:emailAddress": "mariobros@dit.com",
            "dit:team": Object {
              "id": "dit:DataHubTeam:3a48318c-9698-e211-a939-e4115bead28a",
              "name": "Digital Data Hub - Live Service",
              "type": Array [
                "Group",
                "dit:Team",
              ],
            },
            "id": "dit:DataHubAdviser:2dc3c098-2f51-43c5-8767-752e8e5065b5",
            "name": "Mario Bros",
            "type": Array [
              "Person",
              "dit:Adviser",
            ],
          },
        ],
        "dit:archived": false,
        "dit:communicationChannel": Object {
          "name": "Email/Website",
        },
        "dit:service": Object {
          "name": "Account Management",
        },
        "dit:status": "complete",
        "dit:subject": "Invited to Technology Roundtable 95",
        "id": "dit:DataHubInteraction:a066cc49-b679-4cbe-93c5-3ceba2f067ec",
        "startTime": "2019-05-02T08:43:40.904649+00:00",
        "type": Array [
          "dit:Event",
          "dit:Interaction",
        ],
        "url": "https://example.com/companies/a4ead59f-dfc8-e511-a5e6-e4115bead28a/interactions/a066cc49-b679-4cbe-93c5-3ceba2f067ec",
      },
      "published": "2019-08-06T09:43:40.904+01:00",
      "type": "Announce",
    },
  ],
  "error": false,
  "hasMore": true,
  "isLoading": false,
  "offset": 20,
  "total": 100,
}
`;

exports[`ActivityFeedApp renders activity feed 1`] = `
Object {
  "activities": Array [
    Object {
      "generator": Object {
        "name": "dit:dataHub",
        "type": "Application",
      },
      "id": 99,
      "object": Object {
        "attributedTo": Array [
          Object {
            "dit:companiesHouseNumber": "987654321",
            "dit:dunsNumber": "123456789",
            "id": "dit:DataHubCompany:a4ead59f-dfc8-e511-a5e6-e4115bead28a",
            "name": "Marco Test",
            "type": Array [
              "Organization",
              "dit:Company",
            ],
          },
          Object {
            "dit:emailAddress": "john@example.com",
            "id": "dit:DataHubContact:04cadecd-2de4-4cda-9fda-59db93495dba",
            "name": "John Doe",
            "type": Array [
              "Person",
              "dit:Contact",
            ],
          },
          Object {
            "dit:emailAddress": "mariobros@dit.com",
            "dit:team": Object {
              "id": "dit:DataHubTeam:3a48318c-9698-e211-a939-e4115bead28a",
              "name": "Digital Data Hub - Live Service",
              "type": Array [
                "Group",
                "dit:Team",
              ],
            },
            "id": "dit:DataHubAdviser:2dc3c098-2f51-43c5-8767-752e8e5065b5",
            "name": "Mario Bros",
            "type": Array [
              "Person",
              "dit:Adviser",
            ],
          },
        ],
        "dit:archived": false,
        "dit:communicationChannel": Object {
          "name": "Email/Website",
        },
        "dit:service": Object {
          "name": "Account Management",
        },
        "dit:status": "complete",
        "dit:subject": "Invited to Technology Roundtable 99",
        "id": "dit:DataHubInteraction:a066cc49-b679-4cbe-93c5-3ceba2f067ec",
        "startTime": "2019-05-02T08:43:40.904649+00:00",
        "type": Array [
          "dit:Event",
          "dit:Interaction",
        ],
        "url": "https://example.com/companies/a4ead59f-dfc8-e511-a5e6-e4115bead28a/interactions/a066cc49-b679-4cbe-93c5-3ceba2f067ec",
      },
      "published": "2019-08-10T09:43:40.904+01:00",
      "type": "Announce",
    },
    Object {
      "generator": Object {
        "name": "dit:dataHub",
        "type": "Application",
      },
      "id": 98,
      "object": Object {
        "attributedTo": Array [
          Object {
            "dit:companiesHouseNumber": "987654321",
            "dit:dunsNumber": "123456789",
            "id": "dit:DataHubCompany:a4ead59f-dfc8-e511-a5e6-e4115bead28a",
            "name": "Marco Test",
            "type": Array [
              "Organization",
              "dit:Company",
            ],
          },
          Object {
            "dit:emailAddress": "john@example.com",
            "id": "dit:DataHubContact:04cadecd-2de4-4cda-9fda-59db93495dba",
            "name": "John Doe",
            "type": Array [
              "Person",
              "dit:Contact",
            ],
          },
          Object {
            "dit:emailAddress": "mariobros@dit.com",
            "dit:team": Object {
              "id": "dit:DataHubTeam:3a48318c-9698-e211-a939-e4115bead28a",
              "name": "Digital Data Hub - Live Service",
              "type": Array [
                "Group",
                "dit:Team",
              ],
            },
            "id": "dit:DataHubAdviser:2dc3c098-2f51-43c5-8767-752e8e5065b5",
            "name": "Mario Bros",
            "type": Array [
              "Person",
              "dit:Adviser",
            ],
          },
        ],
        "dit:archived": false,
        "dit:communicationChannel": Object {
          "name": "Email/Website",
        },
        "dit:service": Object {
          "name": "Account Management",
        },
        "dit:status": "complete",
        "dit:subject": "Invited to Technology Roundtable 98",
        "id": "dit:DataHubInteraction:a066cc49-b679-4cbe-93c5-3ceba2f067ec",
        "startTime": "2019-05-02T08:43:40.904649+00:00",
        "type": Array [
          "dit:Event",
          "dit:Interaction",
        ],
        "url": "https://example.com/companies/a4ead59f-dfc8-e511-a5e6-e4115bead28a/interactions/a066cc49-b679-4cbe-93c5-3ceba2f067ec",
      },
      "published": "2019-08-09T09:43:40.904+01:00",
      "type": "Announce",
    },
    Object {
      "generator": Object {
        "name": "dit:dataHub",
        "type": "Application",
      },
      "id": 97,
      "object": Object {
        "attributedTo": Array [
          Object {
            "dit:companiesHouseNumber": "987654321",
            "dit:dunsNumber": "123456789",
            "id": "dit:DataHubCompany:a4ead59f-dfc8-e511-a5e6-e4115bead28a",
            "name": "Marco Test",
            "type": Array [
              "Organization",
              "dit:Company",
            ],
          },
          Object {
            "dit:emailAddress": "john@example.com",
            "id": "dit:DataHubContact:04cadecd-2de4-4cda-9fda-59db93495dba",
            "name": "John Doe",
            "type": Array [
              "Person",
              "dit:Contact",
            ],
          },
          Object {
            "dit:emailAddress": "mariobros@dit.com",
            "dit:team": Object {
              "id": "dit:DataHubTeam:3a48318c-9698-e211-a939-e4115bead28a",
              "name": "Digital Data Hub - Live Service",
              "type": Array [
                "Group",
                "dit:Team",
              ],
            },
            "id": "dit:DataHubAdviser:2dc3c098-2f51-43c5-8767-752e8e5065b5",
            "name": "Mario Bros",
            "type": Array [
              "Person",
              "dit:Adviser",
            ],
          },
        ],
        "dit:archived": false,
        "dit:communicationChannel": Object {
          "name": "Email/Website",
        },
        "dit:service": Object {
          "name": "Account Management",
        },
        "dit:status": "complete",
        "dit:subject": "Invited to Technology Roundtable 97",
        "id": "dit:DataHubInteraction:a066cc49-b679-4cbe-93c5-3ceba2f067ec",
        "startTime": "2019-05-02T08:43:40.904649+00:00",
        "type": Array [
          "dit:Event",
          "dit:Interaction",
        ],
        "url": "https://example.com/companies/a4ead59f-dfc8-e511-a5e6-e4115bead28a/interactions/a066cc49-b679-4cbe-93c5-3ceba2f067ec",
      },
      "published": "2019-08-08T09:43:40.904+01:00",
      "type": "Announce",
    },
    Object {
      "generator": Object {
        "name": "dit:dataHub",
        "type": "Application",
      },
      "id": 96,
      "object": Object {
        "attributedTo": Array [
          Object {
            "dit:companiesHouseNumber": "987654321",
            "dit:dunsNumber": "123456789",
            "id": "dit:DataHubCompany:a4ead59f-dfc8-e511-a5e6-e4115bead28a",
            "name": "Marco Test",
            "type": Array [
              "Organization",
              "dit:Company",
            ],
          },
          Object {
            "dit:emailAddress": "john@example.com",
            "id": "dit:DataHubContact:04cadecd-2de4-4cda-9fda-59db93495dba",
            "name": "John Doe",
            "type": Array [
              "Person",
              "dit:Contact",
            ],
          },
          Object {
            "dit:emailAddress": "mariobros@dit.com",
            "dit:team": Object {
              "id": "dit:DataHubTeam:3a48318c-9698-e211-a939-e4115bead28a",
              "name": "Digital Data Hub - Live Service",
              "type": Array [
                "Group",
                "dit:Team",
              ],
            },
            "id": "dit:DataHubAdviser:2dc3c098-2f51-43c5-8767-752e8e5065b5",
            "name": "Mario Bros",
            "type": Array [
              "Person",
              "dit:Adviser",
            ],
          },
        ],
        "dit:archived": false,
        "dit:communicationChannel": Object {
          "name": "Email/Website",
        },
        "dit:service": Object {
          "name": "Account Management",
        },
        "dit:status": "complete",
        "dit:subject": "Invited to Technology Roundtable 96",
        "id": "dit:DataHubInteraction:a066cc49-b679-4cbe-93c5-3ceba2f067ec",
        "startTime": "2019-05-02T08:43:40.904649+00:00",
        "type": Array [
          "dit:Event",
          "dit:Interaction",
        ],
        "url": "https://example.com/companies/a4ead59f-dfc8-e511-a5e6-e4115bead28a/interactions/a066cc49-b679-4cbe-93c5-3ceba2f067ec",
      },
      "published": "2019-08-07T09:43:40.904+01:00",
      "type": "Announce",
    },
    Object {
      "generator": Object {
        "name": "dit:dataHub",
        "type": "Application",
      },
      "id": 95,
      "object": Object {
        "attributedTo": Array [
          Object {
            "dit:companiesHouseNumber": "987654321",
            "dit:dunsNumber": "123456789",
            "id": "dit:DataHubCompany:a4ead59f-dfc8-e511-a5e6-e4115bead28a",
            "name": "Marco Test",
            "type": Array [
              "Organization",
              "dit:Company",
            ],
          },
          Object {
            "dit:emailAddress": "john@example.com",
            "id": "dit:DataHubContact:04cadecd-2de4-4cda-9fda-59db93495dba",
            "name": "John Doe",
            "type": Array [
              "Person",
              "dit:Contact",
            ],
          },
          Object {
            "dit:emailAddress": "mariobros@dit.com",
            "dit:team": Object {
              "id": "dit:DataHubTeam:3a48318c-9698-e211-a939-e4115bead28a",
              "name": "Digital Data Hub - Live Service",
              "type": Array [
                "Group",
                "dit:Team",
              ],
            },
            "id": "dit:DataHubAdviser:2dc3c098-2f51-43c5-8767-752e8e5065b5",
            "name": "Mario Bros",
            "type": Array [
              "Person",
              "dit:Adviser",
            ],
          },
        ],
        "dit:archived": false,
        "dit:communicationChannel": Object {
          "name": "Email/Website",
        },
        "dit:service": Object {
          "name": "Account Management",
        },
        "dit:status": "complete",
        "dit:subject": "Invited to Technology Roundtable 95",
        "id": "dit:DataHubInteraction:a066cc49-b679-4cbe-93c5-3ceba2f067ec",
        "startTime": "2019-05-02T08:43:40.904649+00:00",
        "type": Array [
          "dit:Event",
          "dit:Interaction",
        ],
        "url": "https://example.com/companies/a4ead59f-dfc8-e511-a5e6-e4115bead28a/interactions/a066cc49-b679-4cbe-93c5-3ceba2f067ec",
      },
      "published": "2019-08-06T09:43:40.904+01:00",
      "type": "Announce",
    },
  ],
  "error": false,
  "hasMore": true,
  "isLoading": false,
  "offset": 20,
  "total": 100,
}
`;
